### PR TITLE
Add video recording to tests

### PR DESCRIPTION
**Pull request by Katka92**
### What does this PR do?
**Adding logic for building a test image.**
Currently, if there is already a test image pushed with the corresponding version, the image is not rebuilt in the PR check flow. Adding logic that will check, if there are some changes made in rh-che/dockerfiles/e2e-saas folder. If so, the test image will not be reused but rebuilt.

**Add video recording to tests.**
In previous versions, there were no videos, but every second the screenshot was taken and a screencast was saved in artifacts. Now, in upstream, they removed that logic and added video recording directly to the docker entrypoint. We need to adapt and to set the same recording in our entrypint.

### CODE REVIEW

**Comment by Katka92** [comment]
[test]

**Comment by Katka92** [comment]
[test]

**Comment by Katka92** [comment]
[test]

**Comment by Katka92** [comment]
[test]

File: functional-tests/devscripts/run_tests.sh [hunk 1]
```diff
@@ -97,7 +97,7 @@ if [[ "$PR_CHECK_BUILD" == "true" ]]; then
   #check version in rh-che pom.xml to use correct version of tests.
   getMavenVersion # Checking the maven version for debugging https://github.com/redhat-developer/rh-che/issues/1716
   version=$(getVersionFromPom)
-  if [[ -z "${version}" ]]; then
+  if [[ -z "${version}" || $version = "" ]]; then
```
Comment (ScrewTSW) [inline functional-tests/devscripts/run_tests.sh:100]: Is this really a necessary change?

File: dockerfiles/e2e-saas/docker-entrypoint.sh [hunk 2]
```diff
@@ -165,5 +174,17 @@ fi
 echo "Installing upstream dependency."
 npm --silent i ../../e2e/
 
+# create a folder and record a video
+ffmpeg_path="./report/ffmpeg_report"
```
Comment (ibuziuk) [inline dockerfiles/e2e-saas/docker-entrypoint.sh:178]: would be nice to extract this into the meaningful functions e.g. strartRecording / stopRecording

File: functional-tests/devscripts/run_tests.sh (hunk 1 above)
Comment (ibuziuk) [reply functional-tests/devscripts/run_tests.sh:100]: `if [ -z "${version}"]; then` should do the trick

Comment (Katka92) [reply functional-tests/devscripts/run_tests.sh:100]: Tests are failing if the version is empty. It's probably not caught by the `-z` switch.

Comment (Katka92) [reply functional-tests/devscripts/run_tests.sh:100]: I see, I'll remove it.

Comment (ibuziuk) [reply functional-tests/devscripts/run_tests.sh:100]: you need to have a single `[` in the if statement

File: dockerfiles/e2e-saas/docker-entrypoint.sh (hunk 2 above)
Comment (Katka92) [reply dockerfiles/e2e-saas/docker-entrypoint.sh]: Done.

File: functional-tests/devscripts/run_tests.sh (hunk 1 above)
Comment (Katka92) [reply functional-tests/devscripts/run_tests.sh]: Done.

**Comment by Katka92** [comment]
@ibuziuk @ScrewTSW I've added changes you suggested. You can see the video here [1]. Please note that first ~30 seconds there is a black screen - it is expected as a test need some time to prepare all the stuff (e.g. build the code). 

[1] http://artifacts.ci.centos.org/devtools/rhche/devtools-rh-che-rh-che-prcheck-dev.rdu2c.fabric8.io/2550/e2e_report/report/ffmpeg_report/output.mp4

**Pull request merged**